Document backfill cascade-up and wip refresh caggs (#2692)
timescale/docs: use-timescale/migration/timescaledb-backfill.md

migration.

- **Stage Command:** is used to create copy tasks for hypertable chunks based
  on the specified completion point (`--until`). An optional filter
  (`--filter`) can be used to refine the hypertables and continuous aggregates
  targeted for staging.

  ```sh
  timescaledb-backfill stage --source $SOURCE_DB --target $TARGET_DB --until '2016-01-02T00:00:00' 
  ```

  The tables to be included in the stage can be controlled by providing
  filtering options:

>   `--filter`: this option accepts a schema-qualified hypertable or continuous
>   aggregate view name, and stages only objects which match the filter. The
>   filter can be a regular expression, which is evaluated in the database by
>   PostgreSQL's regular expression engine.

Reviewer: ```suggestion
  `--filter`: this option accepts a POSIX regular expression to match schema-qualified hypertable names or continuous aggregate view names. Only hypertables and/or continuous aggregates matching the filter are staged.
```

  By default, the filter includes only the matching objects, and does not

>   concern itself with dependencies between objects. This is problematic for

Reviewer: ```suggestion
  concern itself with dependencies between objects. Depending on what is intended, this could be problematic for
```

  continuous aggregates, as they form a dependency hierarchy. This behaviour
  can be modified through cascade options.
 
  For example, assuming a hierarchy of continuous aggregates for hourly, daily,
  and weekly rollups of data in an underlying hypertable called `raw_data` (all
  in the `public` schema). This could look as follows:

  ```
  raw_data -> hourly_agg -> daily_agg -> monthly_agg
  ``` 
  

>   If the filter `--filter=public.data` is applied, then no data from the

Reviewer: ```suggestion
  If the filter `--filter='^public\.raw_data$'` is applied, then no data from the
```

  continuous aggregates is staged. If the filter

>   `--filter=public.daily_agg` is applied, then only materialized data in the

Reviewer: ```suggestion
  `--filter='^public\.daily_agg$'` is applied, then only materialized data in the
```

  continuous aggregate `daily_agg` is staged. 

  `--cascade-up`: when activated, this option ensures that any continuous
  aggregates which depend on the filtered object are included in the staging
  process. It is called "cascade up" because it cascades up the hierarchy.
  Using the example from before, if the filter

>   `--filter=public.data --cascade up` is applied, the data in `raw_data`,

Reviewer: ```suggestion
  `--filter='^public\.raw_data$' --cascade up` is applied, the data in `raw_data`,
```

  `hourly_agg`, `daily_agg`, and `monthly_agg` is staged.

  `--cascade-down`: when activated, this option ensures that any objects which
  the filtered object depends on are included in the staging process. It is
  called "cascade down" because it cascades down the hierarchy.
  Using the example from before, if the filter

>   `--filter=public.daily_agg --cascade-down` is applied, the data in

Reviewer: ```suggestion
  `--filter='^public\.daily_agg$' --cascade-down` is applied, the data in
```

  `daily_agg`, `hourly_agg`, and `raw_data` is staged.

  The `--cascade-up` and `--cascade-down` options can be combined. Using the
  example from before, if the filter

>   `--filter=public.daily_agg --cascade-up --cascade-down` is applied, data in

Reviewer: ```suggestion
  `--filter='^public\.daily_agg$' --cascade-up --cascade-down` is applied, data in
```

  all objects in the example scenario is staged.

  ```sh
  timescaledb-backfill stage --source $SOURCE_DB --target $TARGET_DB \
    --until '2016-01-02T00:00:00' \

>     --filter 'public.daily_agg' \

Reviewer: ```suggestion
    --filter '^public\.daily_agg$' \
```

### Usage examples 

- Backfilling with a filter and until date: 

  ```sh
  timescaledb-backfill stage --source $SOURCE_DB --target $TARGET_DB \

>     --filter 'my_table.*' \

Reviewer: ```suggestion
    --filter '.*\.my_table.*' \
```
Not sure what this was supposed to match originally

  timescaledb-backfill stage --source $SOURCE_DB --target $TARGET_DB \

>     --filter 'schema1.table_with_time_as_timestampz' \

Reviewer: ```suggestion
    --filter '^schema1\.table_with_time_as_timestampz$' \
```

    --until '2015-01-01T00:00:00'

  timescaledb-backfill stage --source $SOURCE_DB --target $TARGET_DB \

>     --filter 'schema1.table_with_time_as_bigint' \

Reviewer: ```suggestion
    --filter '^schema1\.table_with_time_as_bigint$' \
```

    --until '91827364'

  timescaledb-backfill stage --source $SOURCE_DB --target $TARGET_DB \

>     --filter 'schema2.*' \

Reviewer: ```suggestion
    --filter '^schema2\..*' \
```

(SIGINT) to the process. This can be achieved by using the Ctrl-C keyboard
shortcut from the terminal where the tool is currently running.

When the tool receives the first signal, it interprets it as a request for a
graceful shutdown. It then notifies the copy workers that they should exit once
they finish copying the chunk they are currently processing. Depending on the
chunk size, this could take many minutes to complete.

When a second signal is received, it forces the tool to shut down immediately,
interrupting all ongoing work. Due to the tool's usage of transactions, there
is no risk of data inconsistency when using forced shutdown. The difference to
a graceful shutdown is that data in partially copied chunks is be removed on

> the target and must be copied again during the next `copy` run.

Reviewer: > While a graceful shutdown waits for in-progress chunks to finish copying, an immediate shutdown rolls back the in-progress copy transactions. Any data copied into those chunks is lost, but the database is left in a transactional consistent state, and the backfill process can be safely resumed.
